packages/webhooks/model-validator/validator.js

```diff
@@ -0,0 +1,208 @@
+        this.fieldsValidators[modelName] = ajv.compile(maskedSchema)
+    }
+
+    /**
+     * Validates specified model filters
+     * @param {string} modelName
+     * @param {JSON} filters
+     * @returns {{isValid: boolean, errors: *[]}}
+     */
+    validateFilters (modelName, filters) {
+        if (!this.models.includes(modelName)) {
+            throw new Error(`Unregistered model name: ${modelName}`)
+        }
+        const validator = this.filterValidators[modelName]
+        const isValid = validator(filters)
+        if (isValid) {
+            return { isValid, errors: [] }
+        } else {
+            return { isValid, errors: validator.errors }
+        }
+    }
+
+    /**
+     * Validates normalized field-string (Subset of model fields in GQL notation)
+     * @param modelName
+     * @param {string} modelName
+     * @param {string} fieldString
+     * @returns {{isValid: boolean, errors: string[]}|{isValid, errors}|{isValid, errors: *[]}}
+     */
+    validateFields (modelName, fieldString) {
+        if (!this.models.includes(modelName)) {
+            throw new Error(`Unregistered model name: ${modelName}`)
+        }
+        const validator = this.fieldsValidators[modelName]
+        const { data, error } = WebHookModelValidator._buildSampleFromFieldsString(fieldString)
+        if (error) {
+            return { isValid: false, errors: [error] }
+        } else {
+            const isValid = validator(data)
+            if (isValid) {
+                return { isValid, errors: [] }
+            } else {
+                return { isValid, errors: validator.errors }
+            }
+        }
+    }
+
+    /**
+     * Wraps field-string in brackets if needed and also fixes spacing issues to make
+     * @param {string} fields
+     * @returns {string}
+     */
+    static normalizeFieldsString (fields) {
+        let trimmedString = fields.trim()
+        if (!trimmedString) {
+            trimmedString = '{}'
+        } else if (!trimmedString.startsWith('{')) {
+            trimmedString = '{' + trimmedString + '}'
+        }
+        const expandedChars = []
+        for (let i = 0; i < trimmedString.length; ++i) {
+
+            if (trimmedString[i] === '{' || trimmedString[i] === '}') {
+                expandedChars.push(' ', trimmedString[i], ' ')
+            } else {
+                expandedChars.push(trimmedString[i])
+            }
+        }
+
+        return expandedChars.join('').trim().split(/\s+/).join(' ')
+    }
+
+    /**
+     * Accepts generated JSON-schema and moves every property types to boolean
+     * Also flattens arrays types and adds minimal amount of specified properties to schema
+     * @param {any} schema
+     * @returns {any}
+     * @private
+     */
+    static _maskSchemaTypes (schema) {
+        if (typeof schema !== 'object') {
+            return schema
+        }
+
+        // NOTE: Deep cloning JSON object
+        const newSchema = JSON.parse(JSON.stringify(schema))
```

Comment:
This manner of cloning keeps only native js values (String, Number, Boolean, ...). Is it ok?

Reply:
This function operates on parts of JSON-schema....so it will cloned without any problems